### PR TITLE
Annotations api

### PR DESCRIPTION
**Pull request by edgarriba**

### CODE REVIEW

**Comment by Hackerman342** [comment]
This should set `main-v2` as the base

File: protos/farm_ng/annotations/annotations.proto [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+// Point2d is a 2D point in image coordinates
+message Point2d {
+  // x coordinate of the point in pixels
+  float x = 1;
+  // y coordinate of the point in pixels
+  float y = 2;
+}
+
+// Image data in raw format wihout any compression
+message Image {
```
[the remark this answers is not in the document]
Comment (edgarriba) [reply protos/farm_ng/annotations/annotations.proto:36]: @ethanrublee i assume these ones ? https://github.com/strasdat/Sophus/blob/sophus2/protos/sophus/image.proto

Can we have in sophus a more python / ml ecosystem friendly proto definitions ? basically as much close as possible to numpy or [dlpack](https://github.com/dmlc/dlpack/blob/main/include/dlpack/dlpack.h) ? /cc @strasdat

[the remark this answers is not in the document]
Comment (strasdat) [reply protos/farm_ng/annotations/annotations.proto:36]: I'm not sure I'm following. The sophus2::DynImage messages contain the required information needed to wrap numpy arrays easily. Indeed, the proposed proto Image above does not carry type and shape information and it is currently hardcoded (see below).

File: py/examples/file_reader_annotations/main.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    # the annotation files have first the image and later the annotation with masks
+    image_log = reader.get_index()[0]
+    annotations_log = reader.get_index()[1:]
+
+    images_vis, titles_vis = [], []
+
+    # visualize the image
+    image = image_log.read_message()
+    image_vis = np.frombuffer(image.data, dtype="uint8")
```
Comment (strasdat) [inline py/examples/file_reader_annotations/main.py:45]: Type is hard-coded here, better to read a from the proto.
In the next line, the number of "channels" is hard-coded to 3. We have that in the sophus2 proto as .pixel_type().num_components()

Comment (edgarriba) [reply py/examples/file_reader_annotations/main.py:45]: How do I define here the number of axis and strides per axis? DynImage seems to only support three dimensional arrays with layout HxWxC, whereas e.g mask here just 2D (HxW). Also depending how, I want CxHxW images when I work with torch and don’t want be permuting axis around, or being able to store batches of images or batches of masks BxHxW. That’s why I was proposing dlpack which in the end is an abstraction of https://numpy.org/doc/stable/reference/c-api/types-and-structures.html#c.PyArrayInterface
Which allows to serialise completely arbitrary data layouts.

File: protos/farm_ng/annotations/annotations.proto [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+
+// Defines a segmentation mask for a single label
+message Annotation {
+    // main label of the annotation; e.g. "tree" or "person"
+    string label = 1;
+    // sublabel of the annotation; e.g. "positive" or "negative"
+    string sublabel = 2;
+    // list of points in image coordinates that define the line
+    repeated sophus.proto.Vec2F32 points = 3;
```
Comment (edgarriba) [inline protos/farm_ng/annotations/annotations.proto:30]: @ethanrublee @strasdat any suggestion whether we should provide normalized coordinates (-1, 1) ?

**Review by Hackerman342 — APPROVED** [review]
LGTM

**Pull request merged**